### PR TITLE
Match calc partner damage enemy

### PR DESCRIPTION
**Pull request by ThreeMustyFears**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.16%) on `us`.

⚠️ This PR introduces 1983 warnings:
See log for details.

File: src/1AC760.c [hunk 1]
```diff
@@ -141,8 +145,652 @@ s32 calc_partner_test_enemy(void) {
...
+                }
+
+                isFireDamage = TRUE;
+            }
+
+            do {    // TODO remove this do while
+                if (gBattleStatus.flags1 & BS_FLAGS1_SP_EVT_ACTIVE && battleStatus->lastAttackDamage >= 0 && dispatchEvent != EVENT_DEATH &&
+                    dispatchEvent != EVENT_SPIN_SMASH_DEATH &&
+                    dispatchEvent != EVENT_EXPLODE_TRIGGER && !(targetPart->targetFlags & 4)) {
+                    #define INFLICT_STATUS(STATUS_TYPE) \
```
Comment (nanaian) [inline src/1AC760.c:550]: Good idea 👍

File: src/1AC760.c [hunk 2]
```diff
@@ -141,8 +145,652 @@ s32 calc_partner_test_enemy(void) {
...
+
+    if (battleStatus->lastAttackDamage <= 0 && !tempBinary && !wasStatusInflicted || targetPart->flags & ACTOR_PART_FLAG_2000) {
+        sfx_play_sound_at_position(SOUND_IMMUNE, 0, state->goalPos.x, state->goalPos.y, state->goalPos.z);
+    }
+
+    if (battleStatus->currentAttackStatus & STATUS_FLAG_SLEEP && wasStatusInflicted) {
+        evt = start_script((EvtScript*) DoSleepHit, 0xA, 0);
+        evt->varTable[0] = (s32) state->goalPos.x;
+        evt->varTable[1] = (s32) state->goalPos.y;
+        evt->varTable[2] = (s32) state->goalPos.z;
```
Comment (nanaian) [inline src/1AC760.c:713]: Are all these casts required?

**Review by nanaian — APPROVED** [review]
Bravo

File: src/1AC760.c (hunk 2 above)
Comment (ThreeMustyFears) [reply src/1AC760.c]: I don't know why I thought they were necessary

File: src/1AC760.c (hunk 1 above)
Comment (ThreeMustyFears) [reply src/1AC760.c:550]: It wasn't mine I stole it from calc_item_damage_enemy

File: src/1AC760.c [hunk 3]
```diff
@@ -141,8 +145,652 @@ s32 calc_partner_test_enemy(void) {
...
+    target = get_actor(currentTargetID);
+    state = &partner->state;
+
+    if (target == NULL) {
+        return 0;
+    }
+
+    targetPart = get_actor_part(target, currentTargetPartID);
+
+    if (targetPart == NULL) {
```
Comment (ethteck) [inline src/1AC760.c:182]: This should probably instead be ASSERT(targetPart != NULL) but if I don't find anytihng else, I'll just merge it and fix it later

**Review by ethteck — APPROVED** [review]

**Pull request merged**